pilot/pkg/model/push_context.go

```diff
@@ -530,6 +530,30 @@ func (ps *PushContext) SubsetToLabels(subsetName string, hostname Hostname) Labe
 	return nil
 }
 
+// SubsetToLabels returns the labels associated with a subset of a given service.
+func (ps *PushContext) SubsetToLabelsIsolated(proxy *Proxy, subsetName string, hostname Hostname) LabelsCollection {
```

Comment:
I dont think you need a new function. The destination rule selection is backward compatible. So older proxies will still select the right destination rule [whether its for EDS or CDS]